### DISCUSSION TITLE
Missing Version and Build number iOS

### DISCUSSION THREAD

**Discussion by SEGVeenstra** [codemagic-ci-cd/codemagic-docs#2182] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Create a Flutter Project
2. Setup workflow (UI) with Firebase Distribution for iOS
3. Run workflow

### Expected results

I expect the build to have the version I've configured in the project's `pubspec.yaml` as the xcode project is configured to use both `$(FLUTTER_BUILD_NUMBER)` and `$(FLUTTER_BUILD_NUMBER)`.

### Actual results

<img width="668" alt="Schermafbeelding 2023-11-08 om 20 06 26" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/5990148/bc7c1109-d45b-4e2e-aac6-ac04c22cc62b">

### Build id (optional)

654bd49309ee3dd469791bf7

**Comment by dtrdic** [comment]
hey @SEGVeenstra, you are missing build arguments for setting build name and build number, like for example here;
`--build-name=1.2.$PROJECT_BUILD_NUMBER --build-number=$(($PROJECT_BUILD_NUMBER + 1))`

- The PROJECT_BUILD_NUMBER variable is a built-in environment variable that represents the current build number of your project.
- It is automatically incremented with each new build of your app.

![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/7e7f4aa1-8015-4476-979c-f966ff7cd2f0)


 Also sure that Info.plist file has the following string along with their key values

```
	<key>CFBundleShortVersionString</key>
	<string>$(FLUTTER_BUILD_NAME)</string>
	<key>CFBundleVersion</key>
	<string>$(FLUTTER_BUILD_NUMBER)</string>
```

**Reply by SEGVeenstra** [reply]
Thank you for your reply.

But I don't want to 'pass' a build name/number, I want it to use the ones specified in my `pubspec.yaml`.

The `Info.plist` does contain the FLUTTER vars, however I recently found out that at some point, these FLUTTER var are no longer present in the Xcode project (under user defined settings). 😕 

So I'm currently setting up the iOS project from scratch, configuring everything again one step at the time, to figure out at what point these disappear. (Because I don't remove them myself)

**Reply by SEGVeenstra** [reply]
Update:
SO far I've added in parts of the configuration and the FLUTTER vars are still there. Will continue adding more of the configiration I originally had to see if at some point it breaks it again.

**Reply by SEGVeenstra** [reply, marked as answer]
I don't know why the Flutter-related vars where missing in my old project, but I've now set up the new iOS project with everything like flavors etc and it is working.
It's getting the version/build information just as I expect.
